providers/aws/resources/aws_acm.go

```diff
@@ -0,0 +1,172 @@
+		regionVal := region
+		f := func() (jobpool.JobResult, error) {
+			svc := conn.Acm(regionVal)
+			ctx := context.Background()
+			res := []interface{}{}
+
+			nextToken := aws.String("no_token_to_start_with")
+			params := &acm.ListCertificatesInput{}
+			for nextToken != nil {
+				certs, err := svc.ListCertificates(ctx, params)
+				if err != nil {
+					if Is400AccessDeniedError(err) {
+						log.Warn().Str("region", regionVal).Msg("error accessing region for AWS API")
+						return res, nil
+					}
+					return nil, err
+				}
+				for _, cert := range certs.CertificateSummaryList {
+					mqlCert, err := a.MqlRuntime.CreateResource(a.MqlRuntime, "aws.acm.certificate",
+						map[string]*llx.RawData{
+							"arn": llx.StringData(toString(cert.CertificateArn)),
+						})
+					if err != nil {
+						return nil, err
+					}
+					res = append(res, mqlCert)
+				}
+				nextToken = certs.NextToken
+				if certs.NextToken != nil {
+					params.NextToken = nextToken
+				}
+			}
+			return jobpool.JobResult(res), nil
+		}
+		tasks = append(tasks, jobpool.NewJob(f))
+	}
+	return tasks
+}
+
+func (a *mqlAwsAcmCertificate) id() (string, error) {
+	return a.Arn.Data, nil
+}
+
+func initAwsAcmCertificate(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if len(args) > 2 {
+		return args, nil, nil
+	}
+
+	if args["arn"] == nil {
+		return nil, nil, errors.New("arn required to fetch aws acm certificate")
+	}
+
+	arnVal := args["arn"].Value.(string)
+	region, err := GetRegionFromArn(arnVal)
+	if err != nil {
+		return args, nil, nil
+	}
+	conn := runtime.Connection.(*connection.AwsConnection)
+
+	svc := conn.Acm(region)
+	ctx := context.Background()
+	certDetails, err := svc.DescribeCertificate(ctx, &acm.DescribeCertificateInput{CertificateArn: &arnVal})
+	if err != nil {
+		return nil, nil, err
+	}
+	certTags, err := svc.ListTagsForCertificate(ctx, &acm.ListTagsForCertificateInput{CertificateArn: &arnVal})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	args["arn"] = llx.StringData(arnVal)
+	args["notBefore"] = llx.TimeData(toTime(certDetails.Certificate.NotBefore))
+	args["notAfter"] = llx.TimeData(toTime(certDetails.Certificate.NotAfter))
+	args["createdAt"] = llx.TimeData(toTime(certDetails.Certificate.CreatedAt))
+	args["domainName"] = llx.StringData(toString(certDetails.Certificate.DomainName))
+	args["status"] = llx.StringData(string(certDetails.Certificate.Status))
+	args["subject"] = llx.StringData(toString(certDetails.Certificate.Subject))
+	args["tags"] = llx.MapData(CertTagsToMapTags(certTags.Tags), types.String)
+	return args, nil, nil
+}
+
+func CertTagsToMapTags(tags []acmtypes.Tag) map[string]interface{} {
+	mapTags := make(map[string]interface{})
+	for i := range tags {
+		if tags[i].Key != nil && tags[i].Value != nil {
+			mapTags[*tags[i].Key] = *tags[i].Value
+		}
+	}
+	return mapTags
+}
+
+// func (a *mqlAwsAcmCertificate) certificate() (*mqlCertificate, error) {
```

Comment:
Is this one of the resources which isn't working yet?

Reply:
ya, the acm certificate stuff -- it was relying on a resource that's available in core and i need to see how that all will work now